lib/engine/game/g_1848/share_pool.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require_relative '../../share_pool'
+
+module Engine
+  module Game
+    module G1848
+      class SharePool < Engine::SharePool
```

Comment:
let's avoid having a share pool if possible, perhaps by having params on the pool or the corporation

Reply:
by adding params to pool or corporation you mean the base game files?
Or copy corporation into g_1848 and modify it there?

Reply:
https://github.com/tobymao/18xx/blob/master/lib/engine/game/g_18_ireland/share_pool.rb, 
does the exact thing i'm after. One corporation (in ireland it's a class of corporation), that has a different market limit.

Is it because it's one company in 1848 you prefer I avoid overwriting share pool?

Reply:
Would this work?:
in https://github.com/tobymao/18xx/blob/master/lib/engine/share_pool.rb#L153 
I change @game.class::MARKET_SHARE_LIMIT into @game.market_share_limit(corporation)
in the base game def market_share_limit(corporation) to return MARKET_SHARE_LIMIT, and in 1848 game.rb i overwrite this method to return the company specific share limit, in this case 100 for boe, 50 for rest?

Reply:
done, as discussed in slack.

Reply:
rookie mistake, didn't run test prior to checkin in my changes. re-trying locally